src/cli/index.js

```diff
@@ -16,57 +16,52 @@ const {
   fetchLibraryVersions,
   getTemplatesByCategory,
   getTemplatePath,
+  splitArray,
 } = require('../utils');
-const getOptionsFromArguments = require('./getOptionsFromArguments');
 const getAttributesFromIndex = require('./getAttributesFromIndex');
 const getFacetsFromIndex = require('./getFacetsFromIndex');
 const getAnswersDefaultValues = require('./getAnswersDefaultValues');
 const isQuestionAsked = require('./isQuestionAsked');
-const {
-  getConfiguration,
-  getLibraryVersion,
-  createNameAlternatives,
-} = require('./getConfiguration');
+const getConfiguration = require('./getConfiguration');
+const postProcessAnswers = require('./postProcessAnswers');
 const { version } = require('../../package.json');
 
 let appPathFromArgument;
-let options = {};
 
 program
   .version(version, '-v, --version')
+  .name('create-instantsearch-app')
   .arguments('<project-directory>')
   .usage(`${chalk.green('<project-directory>')} [options]`)
-  .option('--name <name>', 'The name of the application or widget')
+  // name is a special case in Commander, if it doesn't have a default and not given, the value becomes a function
+  .option('--name <name>', 'The name of the application or widget', '')
```

Comment:
Why don't we pass `undefined` so that we use the same code path for this option as the other ones? (does Commander override `undefined`?)

```suggestion
  .option('--name <name>', 'The name of the application or widget', undefined)
```

Reply:
Quickly going through Commander's doc, it seems like [`helpOption`](https://github.com/tj/commander.js#helpoptionflags-description) was made for this.

Reply:
when I tried undefined it still gave a function and not undefined. Can try it out more

Reply:
so I'm not fully sure yet since which version of Commander it's fixed, but in the latest version the properties are no longer saved on the main object, but in a separate object. 

https://github.com/tj/commander.js/blob/43f4743864e2f670db5eebcf88c92aa4612c54f1/lib/command.js#L1438

Before this was fixed, it caused "name" to be the program name, and not the argument name. It's solvable by renaming --name to -app-name, but that wouldn't really work. Setting a real value as the fallback will prevent it from being a function (the value of .name), but not really sure how to solve properly.

Updating to latest commander should work but it only supports node 12, are we ok with that?

Reply:
https://github.com/tj/commander.js/issues/1226#issuecomment-602176211